### PR TITLE
Add explicit gserver.site_name string truncate in Model\GServer

### LINKED ISSUE #11638: Data too long for column 'site_name'
From the log of Friendica 2022.06:

```
[ERROR]: DB Error {
    "code": 1406,
    "error": "Data too long for column 'site_name' at row 1",
    "callstack": "DBA::insert, GServer::detect, GServer::check, GServer::getID, APContact::getByURL, HTTPSignature::fetchKey, HTTPSignature::getSigner, Followers::rawContent",
    "params":"INSERT INTO `gserver` (`detection-method`, `register_policy`, `platform`, `version`, `site_name`, `network`, `noscrape`, `directory-type`, `url`, `nurl`, `registered-users`, `next_contact`, `last_contact`, `failed`, `created`) VALUES (102, 2, 'friendica', '2022.05-rc', 'Ibbtown.com    ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  ⠀  <br><br>  <b><a href=\\\"https://ibbtown.com/display/2338580c-1962-7bf4-1ee6-f5f584362672\\\">Was ist das hier?</a></b> | <b><a href=\\\"https://ibbtown.com/community/\\\">Öffentliche Zeitleiste</a></b> | <b>Admin: @<a href=\\\"https://ibbtown.com/profile/ibbtown\\\">ibbtown</a></b> <br><br><b><a href=\\\"https://ibb.town\\\">ibb.town</a></b> | <b><a href=\\\"https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=7DPJ7XTJZ4UGW\\\">Kaffeekasse</a></b> | <b><a href=\\\"https://ibb.town/kontakt\\\">Kontakt & Datenschutzerklärung</a></b>', 'dfrn', 'https://ibbtown.com/noscrape', 2, 'https://ibbtown.com', 'http://ibbtown.com', 1, '2022-06-19 14:30:25', '2022-06-12 14:30:25', 0, '2022-06-12 14:30:25')"
} - {
    "file": "Database.php",
    "line": 782,
    "function": "e",
    "uid": "b346e9",
    "process_id": 77702
}
```

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #11638 

- Avoids "Data too long for column 'site_name'" database errors

### CODE REVIEW

File: src/Model/GServer.php [hunk 1]
```diff
@@ -2063,4 +2063,24 @@ public static function getProtocol(int $gsid): ?int
...
+	 * @param array $condition
+	 * @return bool
+	 * @throws Exception
+	 */
+	public static function update(array $fields, array $condition): bool
+	{
+		/* These single byte character string functions are intended to support systems where the size of the
+		 * VARCHAR field is expressed in bytes (MySQL 4) and not in characters (MySQL 5, MariaDB)
+		 */
+		if (isset($fields['site_name']) && strlen($fields['site_name']) > 255) {
```
Comment (annando) [inline src/Model/GServer.php]: Just use:
```
$fields = DBStructure::getFieldsForTable('gserver', $fields);
```
That does all the job.

Comment (MrPetovan) [reply src/Model/GServer.php]: Neat!

File: src/Model/GServer.php [hunk 2]
```diff
@@ -2063,4 +2064,19 @@ public static function getProtocol(int $gsid): ?int
...
+	 * Enforces gserver table field maximum sizes to avoid "Data too long" database errors
+	 *
+	 * @param array $fields
+	 * @param array $condition
+	 * @return bool
+	 * @throws Exception
+	 */
+	public static function update(array $fields, array $condition): bool
+	{
+		$fields = DBStructure::getFieldsForTable('post-thread', $fields);
```
Comment (annando) [inline src/Model/GServer.php]: You should use "gserver" and not "post-thread", because you want to match against the definition of that table.

Comment (MrPetovan) [reply src/Model/GServer.php]: Copy-paste issue, you can relate!

**Pull request merged**